src/asio_service.cxx

```diff
@@ -1590,13 +1590,18 @@ std::string asio_service_impl::get_password
 #endif
 
 void asio_service_impl::worker_entry() {
-    std::string thread_name = "nuraft_w_" + std::to_string(worker_id_.fetch_add(1));
+    uint32_t worker_id = worker_id_.fetch_add(1);
+    std::string thread_name = "nuraft_w_" + std::to_string(worker_id);
 #ifdef __linux__
     pthread_setname_np(pthread_self(), thread_name.c_str());
 #elif __APPLE__
     pthread_setname_np(thread_name.c_str());
 #endif
 
+    if (my_opt_.worker_start_) {
+        my_opt_.worker_start_(worker_id);
```

Comment:
Not sure `worker_id` is what you intended (it is different from thread ID). Let me merge it for now.